R/package.R

```diff
@@ -443,8 +443,8 @@ renv_package_unpack <- function(package, path, subdir = "", force = FALSE) {
   }
 
   # create extraction directory
-  old <- tempfile("renv-package-old-")
-  new <- tempfile("renv-package-new-")
+  old <- renv_scope_tempfile("renv-package-old-")
+  new <- renv_scope_tempfile("renv-package-new-", envir = parent.frame())
```

Comment:
any reason why we need the explicit environment here but not above?

Reply:
Because the function returns `new`, so we want to make sure it's available to the caller. This might be too magical, and it might be better to not cleanup at all here.